planner/core/logical_plan_builder.go

```diff
@@ -986,7 +986,11 @@ func (b *PlanBuilder) buildSelection(ctx context.Context, p LogicalPlan, where a
 		for _, item := range cnfItems {
 			if con, ok := item.(*expression.Constant); ok && con.DeferredExpr == nil && con.ParamMarker == nil {
 				ret, _, err := expression.EvalBool(b.ctx, expression.CNFExprs{con}, chunk.Row{})
```

Comment:
Seems the `EvalBool` behavior is not the same with MySQL
```
mysql> /* s1 */ select * from t where ' ';
Empty set, 1 warning (0.00 sec)

mysql> show warnings;
+---------+------+--------------------------------------+
| Level   | Code | Message                              |
+---------+------+--------------------------------------+
| Warning | 1292 | Truncated incorrect DOUBLE value: '' |
+---------+------+--------------------------------------+
1 row in set (0.01 sec)
```

```
mysql> /* s1 */ select * from t where ' ';
Empty set (0.00 sec)
```

The first one is the TiDB's. You can see that MySQL doesn't throw warning out.
This could be a problem if we use a strict SQL mode. MySQL would success and TiDB would fail